update_external_sources.bat

```diff
@@ -21,3 +21,4 @@ git submodule update --init --recursive
       echo JsonCPP Amalgamation failed to generate %JSONCPP_DIR%\dist\json\json.h
       set errorCode=1
    )
+   cd %BUILD_DIR%
```

Comment:
Just curious: does this actually do something in a .bat file?  I.e. a .bat file runs in the same environment as its caller, and can change said environment?

Reply:
Yes, a .bat file runs in the same environment as the caller. Without this final cd command, the script will exit and the caller will have a new cwd. This would have necessitated adding a cd command back up a few levels in the set of instructions in the BUILD.md file.